### DISCUSSION TITLE
Declare the type of a class field to be the type of another class field

### DISCUSSION THREAD

**Discussion by leehangwee** [pydantic/pydantic#2909] [Question] [answered]
Is it possible to declare the type of a class field to be the type of another class field?
In the example below, I would like to declare the field type of `first_name` from class `UpdateUserDetails` to be based on the field type of `first_name` from the class `User`, so that I dont have to repeat the field type twice.

```
from uuid import UUID

class User(BaseModel):
    name: Union[str, None] = None
    id: UUID
    first_name: str
    last_name: str

class UpdateUserDetails(BaseModel):
    first_name: User.first_name
```

If this is not possible, how best can I implement this to enable better code reuse?

**Comment by PrettyWood** [comment, marked as answer]
Hi @hwlee96 
You can use [`typing.get_type_hints`](https://docs.python.org/3.10/library/typing.html#typing.get_type_hints) or directly `__annotations__` like this
```py
class UpdateUserDetails(BaseModel):
    first_name: User.__annotations__['first_name']
```
Hope it helps!

**Reply by leehangwee** [reply]
Thank you!
